### PR TITLE
Fix jekyll build

### LINKED ISSUE #1111: Docs fail to build in Vagrant
The docs install script produces the following error:

```
==> ubuntu-bionic: Running provisioner: build-site-jekyll (shell)...
    ubuntu-bionic: Running: /tmp/vagrant-shell20201118-171947-ilj452.sh
    ubuntu-bionic: Installing Ruby
    ubuntu-bionic: Reading package lists...
    ubuntu-bionic: Building dependency tree...
    ubuntu-bionic: Reading state information...
    ubuntu-bionic: build-essential is already the newest version (12.4ubuntu1).
    ubuntu-bionic: zlib1g-dev is already the newest version (1:1.2.11.dfsg-0ubuntu2).                                                                                                          
    ubuntu-bionic: zlib1g-dev set to manually installed.                                                                                                                                       
    ubuntu-bionic: The following additional packages will be installed:                                                                                                                        
    ubuntu-bionic:   fonts-lato javascript-common libgmp-dev libgmpxx4ldbl libjs-jquery
    ubuntu-bionic:   libruby2.5 rake ri ruby ruby-dev ruby-did-you-mean ruby-minitest
    ubuntu-bionic:   ruby-net-telnet ruby-power-assert ruby-test-unit ruby2.5 ruby2.5-dev
    ubuntu-bionic:   ruby2.5-doc rubygems-integration
    ubuntu-bionic: Suggested packages:
    ubuntu-bionic:   gmp-doc libgmp10-doc libmpfr-dev bundler
    ubuntu-bionic: The following NEW packages will be installed:
    ubuntu-bionic:   fonts-lato javascript-common libgmp-dev libgmpxx4ldbl libjs-jquery
    ubuntu-bionic:   libruby2.5 rake ri ruby ruby-dev ruby-did-you-mean ruby-full ruby-minitest
    ubuntu-bionic:   ruby-net-telnet ruby-power-assert ruby-test-unit ruby2.5 ruby2.5-dev
    ubuntu-bionic:   ruby2.5-doc rubygems-integration
    ubuntu-bionic: 0 upgraded, 20 newly installed, 0 to remove and 36 not upgraded.
    ubuntu-bionic: Need to get 8,366 kB of archives.                                                                                                                                           
    ubuntu-bionic: After this operation, 48.1 MB of additional disk space will be used.                                                                                                        
    ubuntu-bionic: Do you want to continue?                                                                                                                                                    
    ubuntu-bionic:  [Y/n] 
    ubuntu-bionic: Abort.
    ubuntu-bionic: Installing Jekyll
    ubuntu-bionic: /tmp/vagrant-shell: line 12: cd: docs: No such file or directory
    ubuntu-bionic: /tmp/vagrant-shell: line 13: gem: command not found
    ubuntu-bionic: /tmp/vagrant-shell: line 16: bundle: command not found
    ubuntu-bionic: Visit http://localhost:8888 to see the LibreTime website.
```

### PR DESCRIPTION
**Pull request by zklosko**
Hopefully fixes #1111

### CODE REVIEW

**Comment by paddatrapper** [comment]
This still has the travis stuff :)

**Comment by paddatrapper** [comment]
Vagrant build fails with

```
There are errors in the configuration of this machine. Please fix
the following errors and try again:

shell provisioner:
* `path` for shell provisioner does not exist on the host system: /home/kyle/projects/libretime/libretime/docs/jekyll.sh
```

I think you need to update Vagrantfile L72

**Comment by paddatrapper** [comment]
```
debian-buster: /tmp/vagrant-shell: line 12: cd: docs: No such file or directory
```

This script seems to run with a working directory of `/home/vagrant/`. In vagrant the docs are in `/vagrant/docs`

**Comment by zklosko** [comment]
Added a user check to the script:

```
# Picking different directory if installing with Vagrant
localUser=$(who am i | awk '{print $1}')
if [ $localUser == vagrant ]
then 
	cd /vagrant/docs
else
	cd docs
fi
```

How about now?

**Comment by paddatrapper** [comment]
Will test. With a manual `cd /vagrant/docs` the script fails:


    debian-buster: Configuration file: /vagrant/docs/_config.yml
    debian-buster:             Source: /vagrant/docs
    debian-buster:        Destination: /vagrant/docs/_site
    debian-buster:  Incremental build: disabled. Enable with --incremental
    debian-buster:       Generating... 
    debian-buster:                     done in 142.257 seconds.
    debian-buster:  Auto-regeneration: enabled for '/vagrant/docs'
    debian-buster:                     ------------------------------------------------
    debian-buster:       Jekyll 4.1.1   Please append `--trace` to the `serve` command 
    debian-buster:                      for any additional information or backtrace. 
    debian-buster:                     ------------------------------------------------

**Comment by zklosko** [comment]
The output you posted indicated a successful run...

**Comment by paddatrapper** [comment]
```
debian-buster:                     ------------------------------------------------
debian-buster:       Jekyll 4.1.1   Please append `--trace` to the `serve` command 
debian-buster:                      for any additional information or backtrace. 
debian-buster:                     ------------------------------------------------
```

This is in red and I can't open http://localhost:8888 in my browser - Unable to connect

**Comment by zklosko** [comment]
That has to do with the network routing with Vagrant and Virturalbox. I've specified the hostname Jekyll should serve on: `0.0.0.0`.

How about now?

**Comment by paddatrapper** [comment]
> How about now?

Same output as before. After that the command exists with a `1` as the exit code

**Comment by paddatrapper** [comment]
> TBH, a Ruby + Jekyll setup is probably going to be easier and faster than using Vagrant to check changes to the docs...

I'm not so sure... All vagrant does is run the `jekyll.sh` script in a contained environment. The long part is waiting for it to finish generating the site

**Comment by paddatrapper** [comment]
Found the error using `--trace`:

```
Traceback (most recent call last):
        24: from /usr/local/bin/jekyll:23:in `<main>'
        23: from /usr/local/bin/jekyll:23:in `load'
        22: from /var/lib/gems/2.5.0/gems/jekyll-4.1.1/exe/jekyll:15:in `<top (required)>'
        21: from /var/lib/gems/2.5.0/gems/mercenary-0.4.0/lib/mercenary.rb:21:in `program'
        20: from /var/lib/gems/2.5.0/gems/mercenary-0.4.0/lib/mercenary/program.rb:44:in `go'
        19: from /var/lib/gems/2.5.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `execute'
        18: from /var/lib/gems/2.5.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `each'
        17: from /var/lib/gems/2.5.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `block in execute'
        16: from /var/lib/gems/2.5.0/gems/jekyll-4.1.1/lib/jekyll/commands/serve.rb:86:in `block (2 levels) in init_with_program'
        15: from /var/lib/gems/2.5.0/gems/jekyll-4.1.1/lib/jekyll/command.rb:91:in `process_with_graceful_fail'
        14: from /var/lib/gems/2.5.0/gems/jekyll-4.1.1/lib/jekyll/command.rb:91:in `each'
        13: from /var/lib/gems/2.5.0/gems/jekyll-4.1.1/lib/jekyll/command.rb:91:in `block in process_with_graceful_fail'
        12: from /var/lib/gems/2.5.0/gems/jekyll-4.1.1/lib/jekyll/commands/serve.rb:102:in `process'
        11: from /var/lib/gems/2.5.0/gems/jekyll-4.1.1/lib/jekyll/commands/serve.rb:215:in `start_up_webrick'
        10: from /var/lib/gems/2.5.0/gems/jekyll-4.1.1/lib/jekyll/commands/serve.rb:215:in `new'
         9: from /usr/lib/ruby/2.5.0/webrick/httpserver.rb:47:in `initialize'
         8: from /usr/lib/ruby/2.5.0/webrick/server.rb:108:in `initialize'
         7: from /usr/lib/ruby/2.5.0/webrick/server.rb:127:in `listen'
         6: from /usr/lib/ruby/2.5.0/webrick/utils.rb:65:in `create_listeners'
         5: from /usr/lib/ruby/2.5.0/socket.rb:762:in `tcp_server_sockets'
         4: from /usr/lib/ruby/2.5.0/socket.rb:227:in `foreach'
         3: from /usr/lib/ruby/2.5.0/socket.rb:227:in `each'
         2: from /usr/lib/ruby/2.5.0/socket.rb:764:in `block in tcp_server_sockets'
         1: from /usr/lib/ruby/2.5.0/socket.rb:201:in `listen'
/usr/lib/ruby/2.5.0/socket.rb:201:in `bind': Address already in use - bind(2) for 0.0.0.0:8888 (Errno::EADDRINUSE)
```

**Comment by paddatrapper** [comment]
Busy troubleshooting, but I think I found the issue - old jekyll process left running

**Comment by zklosko** [comment]
@paddatrapper I completely agree. The docs consistently build in under 10 seconds when built on my computer. If a contributor is working on rewording/rewriting they don't necessarily need to do a technical build of the docs; all typos can be checked from the diffs. A full docs build would only be necessary if there would be a site redesign, JS plugin configuration, or checking for bad links (thought that can mostly be done by looking at the files).

I already have a quick blurb on the website about using Jekyll, but I can include more and add a readme to the docs folder in #1101.

**Comment by paddatrapper** [comment]
The explanation could even be here. The important thing is how to get from first git clone to fully built docs. I.e. the packages that need to be installed and commands that need to be run

**Comment by zklosko** [comment]
Absolutely.

So, remove docs from Vagrant? I'd love to help test but I keep getting port errors in Vagrant.

**Comment by paddatrapper** [comment]
Yeah, remove the docs from Vagrant

I can probably try help fix those errors over on discourse or mattermost at some point if you want

**Comment by zklosko** [comment]
You're the best 😄

File: jekyll.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#/bin/bash
+
+echo "Installing Ruby"
+apt-get update -y && apt-get install -y ruby-full build-essential zlib1g-dev
+
+export GEM_HOME=".gems"
```
Comment (paddatrapper) [inline jekyll.sh:6]: This probably needs a .gitignore entry too for `/docs/.gems/`

**Comment by paddatrapper** [comment]
Do you want to add docs to that effect as part of this PR or #1101?

**Comment by zklosko** [comment]
It can go in #1101. I have a few other things I'm writing for that PR too. Also working on a site redesign (with search).

**Pull request merged**